Hw5 + bonus

ViciousXerra opened the pull request.

ViciousXerra committed:
Started

ViciousXerra committed:
Bot RestController completed

ViciousXerra committed:
Scrapper requests/responses and error responses DTO created

ViciousXerra committed:
configuring springdoc

ViciousXerra committed:
trying to bind springdoc to port 8080

ViciousXerra committed:
added custom link validation
created documentation

ViciousXerra committed:
create client interfaces to scrapper api

ViciousXerra committed:
Update scrapper and bot clients
Created DTOs for clients

ViciousXerra committed:
Create config to scrapper clients

ViciousXerra committed:
Create config to bot client

ViciousXerra committed:
Refactoring

ViciousXerra committed:
???

ViciousXerra committed:
Why POST, GET /links legit and DELETE are not?

ViciousXerra committed:
Solved, fixed LinksClient.java

ViciousXerra committed:
Solved, fixed LinksClient.java

ViciousXerra committed:
Fixed checkstyle

ViciousXerra committed:
Fixed checkstyle

ViciousXerra committed:
Fix closing spring context

ViciousXerra committed:
Fix closing spring context

ViciousXerra committed:
Fix closing spring context

ViciousXerra committed:
Create BotClient tests

ViciousXerra committed:
Create BotClient tests

ViciousXerra committed:
Redacted scrapper client tests

ViciousXerra committed:
Redacted bot client tests

ViciousXerra committed:
Create tables,
create master.xml

ViciousXerra committed:
Connection established on 5432 port
Tables has been presented

ViciousXerra committed:
Add BotRestController basic web mvc tests

ViciousXerra committed:
Add TelegramChatController basic web mvc tests

ViciousXerra committed:
Add TelegramChatController basic web mvc tests

ViciousXerra committed:
Tests redacted

ViciousXerra committed:
migrations redacted:
Added a meta-info comments;
Tables modified (IF NOT EXISTS)

ViciousXerra committed:
Tests redacted

ViciousXerra committed:
Tests redacted

ViciousXerra committed:
Tests redacted

ViciousXerra committed:
Tests redacted

ViciousXerra committed:
merge-commit hw3

ViciousXerra committed:
merge-commit hw4

ViciousXerra committed:
BotUpdateClientTest.java fixed (disabled auto configuration)

ViciousXerra committed:
Fixed test classes:
BotUpdateClientTest.java
IntegrationTest.java

Included configurations:
PostgreSQLConfig.java

ViciousXerra committed:
Fixed checkstyle

ViciousXerra committed:
Fixed integration test

ViciousXerra committed:
Redacted Chat.sql schema;
Redacted ChatIdLinkId.sql schema

ViciousXerra committed:
Redacted Chat.sql schema;
Redacted ChatIdLinkId.sql schema

ViciousXerra committed:
Merge branch 'hw4' into hw5

ViciousXerra committed:
Created repository interfaces

ViciousXerra committed:
Created chat repository interface

ViciousXerra committed:
Chat table schema redacted:
removed username (useless)

ViciousXerra committed:
Merge branch 'hw4' into hw5

ViciousXerra committed:
Package hierarchy refactoring
(moved GitHubClient and StackOverFlowClient)

ViciousXerra committed:
Repository interfaces and dto created

ViciousXerra committed:
Repository interfaces and dto created

ViciousXerra committed:
Merge branch 'hw4' into hw5

ViciousXerra committed:
Created DTOs and jdbc repository implementations

ViciousXerra committed:
Redacted dynamic properties

ViciousXerra committed:
Merge branch 'hw4' into hw5

ViciousXerra committed:
Connection pool problem fixed

ViciousXerra committed:
Singleton container issue resloved

ViciousXerra committed:
Created jdbc repositories tests

ViciousXerra committed:
Redacted jdbc repositories tests:
added LinkRepository test

ViciousXerra committed:
Switched Link repository return optional link instead of link

ViciousXerra committed:
Repositories tests completed

ViciousXerra committed:
Repositories tests completed

ViciousXerra committed:
Moved dto package on ../

ViciousXerra committed:
Added tests on cascade deletion

ViciousXerra committed:
Added chat service

ViciousXerra committed:
Refactoring:
Nested LinkValidation annotation in LinkValidator.java
Fixed @LinkValidation in DTOs
Fixed ScrapperClientConfig.java
Fixed BotControllerAdvice.java, ScrapperControllerAdvice.java
Removed "documentation" annotations in REST - controllers

ViciousXerra committed:
Refactoring:
ChatClient.java
LinksClient.java

ViciousXerra committed:
Removed dirties context annotation
Redacted ScrapperClientConfig.java

ViciousXerra committed:
Removed unused imports

ViciousXerra committed:
Tests redacted.

ViciousXerra committed:
Tests redacted.

ViciousXerra committed:
Tests redacted.

ViciousXerra committed:
pulled changed hw3 files

ViciousXerra committed:
ScrapperControllerAdvice.java redacted

ViciousXerra committed:
Scrapper REST controllers refactoring

ViciousXerra committed:
Merge branch 'hw3' into hw5

ViciousXerra committed:
Added service interfaces;
Repository interfaces redacted;
Implementing JdbcLinkRepository.java and JdbcChatRepository.java
Repository tests redacted;

ViciousXerra committed:
DAO services completed.

ViciousXerra committed:
Fixed checkstyle

ViciousXerra committed:
Refactoring LinksRestController.java

ViciousXerra committed:
Fixed BotClientConfig.java

ViciousXerra committed:
Merge branch 'hw3' into hw5

ViciousXerra committed:
Redacted scheduler config;
application.yml;
Created JdbcLinkUpdater.java

ViciousXerra committed:
Fixed unregister method in REST controller

ViciousXerra committed:
Merge branch 'main' into hw3

ViciousXerra committed:
merging main into hw3

ViciousXerra committed:
Merge remote-tracking branch 'java2024/hw3' into hw3

# Conflicts:
#	bot/src/main/resources/application.yml
#	scrapper/src/main/java/edu/java/scrapper/configuration/ApplicationConfig.java

ViciousXerra committed:
merging hw3 into hw5

ViciousXerra committed:
UpdaterScheduler almost done:
created:
Domain enum
Parser

ViciousXerra committed:
UpdaterScheduler almost done:
created:
Domain enum
Parser

ViciousXerra committed:
UpdaterScheduler almost done:
created:
Domain enum
Parser

ViciousXerra committed:
LinkUpdaterScheduler.java redacted:
Added logic based on DAO services and repositories;
Created AbstractLinkResourceUpdater.java and extensions;
Created config class for AbstractLinkResourceUpdater.java
Moved LinkUpdaterUtils.java

ViciousXerra committed:
Created:
JdbcChatServiceTest.java
JdbcLinkServiceTest.java
Redacted:
JdbcChatRepositoryTest.java

ViciousXerra committed:
Created:
JdbcLinkUpdaterTest.java

ViciousXerra committed:
Redacted:
JdbcLinkServiceTest.java
JdbcLinkUpdaterTest.java

ViciousXerra committed:
Redacted:
JdbcLinkServiceTest.java
JdbcLinkUpdaterTest.java

ViciousXerra committed:
Redacted:
Scrapper REST controllers tests

ViciousXerra committed:
Added:
scrapper-jooq module
Generated:
jooq dao entities

ViciousXerra committed:
Fixed:
RESOURCE_SLICER_REGEX in LinkUpdaterUtils.java

ViciousXerra committed:
Added:
jooq repository tests
Regenerated jooq entities

ViciousXerra committed:
Added:
Jooq DAO services + tests
Redacted:
Bean renaming for jdbc dao types (temp)

ViciousXerra committed:
Bonus completed

ViciousXerra committed:
Redacted scheduler and message builder types

ViciousXerra committed:
Fixed checkstyle

ViciousXerra committed:
Fixed checkstyle

ViciousXerra committed:
Fixed checkstyle

Betorov commented on scrapper/src/main/java/edu/java/scrapper/dao/repository/jdbc/JdbcChatIdLinkIdRepository.java:
```diff
@@ -0,0 +1,87 @@
+import org.springframework.jdbc.core.simple.JdbcClient;
+import org.springframework.stereotype.Repository;
+
+@Repository("jdbc-chatidlinkid-repo")
```
Лучше бины название выводить в константы.

Можно прям здесь сделать public static final String CHATID_BEAN_NAME

Betorov commented on scrapper/src/main/java/edu/java/scrapper/domain/jooq/DefaultCatalog.java:
Предлагаю модель данных генерируемых жуком вынести в отдельный модуль. Будет легче потом собирать и манипулировать с ним. 

Изменение, случайно код изменишь который потом перегенириуется и все :)

ViciousXerra replied:
Приветствую, а не подскажете, как это сделать? Насколько я понял, надо создавать новый maven проект и внедрить как зависимость в scrapper. Я попытался так сделать, у меня валится всё, что очевидно, потому что я не знаю, что я делаю.

ViciousXerra committed:
Redacted:
scrapper tests

ViciousXerra committed:
Moved:
Generated source code by jooq in separate module.
Added:
checkstyle-suppressions for scrapper-jooq and jooq-source-codegen modules;
jooq-source-codegen dependency in pom.xml (scrapper)

ViciousXerra committed:
Bean naming by public static final fields

ViciousXerra committed:
Bean naming by public static final fields

ViciousXerra committed:
Bean naming by public static final fields

ViciousXerra committed:
Update pom.xml
